src/detectLaunchConfigurationChanges.ts

```diff
@@ -0,0 +1,32 @@
+import { Workspace } from "./interfaces/Workspace";
+
+interface PromptRestartParams {
+  message: string;
+  reloadWindowChoice: string;
+  dismissChoice: string;
+}
+
+export function detectLaunchConfigurationChanges(
+  workspace: Workspace,
+  promptRestart: (params: PromptRestartParams) => Thenable<void>
+): void {
+  workspace.onDidChangeConfiguration(e => {
+    const promptRestartKeys = [
+      "serverVersion",
+      "serverProperties",
+      "javaHome",
+      "customRepositories"
+    ];
+    const shouldPromptRestart = promptRestartKeys.some(k =>
```

Comment:
nit: can we just use `key` rather than `k`

Reply:
I like explicit names too, done!